quiche/src/recovery/gcongestion/bbr2/startup.rs

```diff
@@ -64,26 +68,47 @@ impl ModeImpl for Startup {
             return Mode::Startup(self);
         }
 
+        let mut startup_exit_reason = None;
+
         let has_bandwidth_growth =
             self.model.has_bandwidth_growth(congestion_event, params);
+        if self.model.full_bandwidth_reached() {
+            startup_exit_reason = Some(StartupExitReason::BandwidthPlateau);
+        }
 
-        #[allow(clippy::absurd_extreme_comparisons)]
-        if params.max_startup_queue_rounds > 0 && !has_bandwidth_growth {
+        let check_persisten_queue =
+            params.max_startup_queue_rounds > 0 && !has_bandwidth_growth;
+        if check_persisten_queue {
             // 1.75 is less than the 2x CWND gain, but substantially more than
             // 1.25x, the minimum bandwidth increase expected during
             // STARTUP.
             self.model.check_persistent_queue(1.75, params);
+            if self.model.full_bandwidth_reached() {
```

Comment:
`if startup_exit_reason.is_none() && self.model.full_bandwidth_reached() {`

Reply:
The none check is now part of the `set_startup_exit` function.